apps/daimo-mobile/src/logic/device.ts

```diff
@@ -0,0 +1,53 @@
+import { isDERPubKey, assert } from "@daimo/common";
+import { Hex, isHex, keccak256 } from "viem";
+
+const emojis = [
+  "🍏",
+  "🍎",
+  "🍐",
+  "🍊",
+  "🍋",
+  "🍌",
+  "🍉",
+  "🍇",
+  "🍓",
+  "🍈",
+  "🍒",
+  "🍑",
+  "🍍",
+  "🥭",
+  "🍅",
+  "🍆",
+  "🥑",
+  "🥝",
+  "🥥",
+  "🥭",
+  "🍠",
+  "🥔",
+  "🥕",
+  "🌽",
+  "🌶️",
+  "🍄",
+  "🥒",
+  "🥦",
+  "🍞",
+  "🥖",
+  "🥐",
+];
+
+export function pubKeyToEmoji(pubKey: Hex): string {
+  return emojis[keccak256(pubKey, "bytes")[0] % emojis.length];
```

Comment:
tough thing with this approach:
- will be fairly common for people to have two `Device 🍄`s
- emojis are distinct style. its worth considering, but if we do it we'll likely end up using emojis elsewhere in the UI too

Reply:
Yeah, I agree, makes sense! Just wanted a stand-in for labelling for now. Given that we want to represent ~256 bits of entropy we should think about a better idea in general.

Reply:
replaced by slots/A/B/C now